library/manageiq.py

```diff
@@ -214,9 +212,9 @@ def main():
             name=dict(required=True),
             type=dict(required=True,
                       choices=['openshift-origin', 'openshift-enterprise']),
-            url=dict(required=True),
-            username=dict(required=True),
-            password=dict(required=True, no_log=True),
+            url=dict(default=os.environ.get('MIQ_URL', None)),
+            username=dict(default=os.environ.get('MIQ_USERNAME', None)),
+            password=dict(default=os.environ.get('MIQ_PASSWORD', None)),
```

Comment:
Don't you also need some logic to enforce "otherwise, it is required to pass it"?
```
if password is None:
    fail_json(...)
``` 
etc.
Or maybe something like:
```
def env_var_or_required(var):
    if var in os.environ:
        return dict(default=os.environ[var])
    else:
        return dict(required=True)

...
    url=env_var_or_required('MIQ_URL'),
    username=env_var_or_required('MIQ_USERNAME'),
    password=env_var_or_required('MIQ_PASSWORD'),
```